### PR TITLE
Updates ESLint rules to error on default imports

### PR DESCRIPTION
**Pull request by katiedavis**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description
Updates our eslint rules to error on default JS exports. This provides consistency when importing javascript across the app. 

## Related Tickets & Documents
See #12383 

## QA Instructions, Screenshots, Recordings
I did some manual sanity testing, but it would be great if someone could just quickly hop on this branch and check to make sure nothing is broken! It most likely isn't though. 

## Added tests?

- [ ] Yes
- [x] No, and this is why: This PR updates a lint rule and the accompanying violations.
- [ ] I need help with writing tests

## Added to documentation?

- [ ] [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] README
- [x] No documentation needed

dancing janitor

### CODE REVIEW

**Review by nickytonline — COMMENTED** [review]
Looking through this and all seems good. I'm just going to pull it locally to take it for a test drive.

**Comment by katiedavis** [comment]
@nickytonline for some reason eslint didn't catch a bunch of files: https://codeclimate.com/github/forem/forem/pull/12512
I'm going to add another commit

File: app/javascript/chat/ChatChannelSettings/RequestedMembershipSection.jsx [hunk 1]
```diff
@@ -39,10 +39,8 @@ const RequestedMembershipSection = ({
 };
 
 RequestedMembershipSection.propTypes = {
-  requestedMemberships: PropTypes.arrayOf(defaulMembershipPropType).isRequired,
+  requestedMemberships: PropTypes.arrayOf(defaultMembershipPropType).isRequired,
```
Comment (katiedavis) [inline app/javascript/chat/ChatChannelSettings/RequestedMembershipSection.jsx:42]: I had to update the typo I couldn't handle it

**Review by aitchiss — APPROVED** [review]
LGTM - I had a quick manual check and couldn't see any issues. I love seeing all those default exports cleaned up 😄

**Review by jacobherrington — APPROVED** [review]
LGTM

:broom: :wastebasket:

**Review by nickytonline — CHANGES_REQUESTED** [review]
Not a request change, just a spot check.

@katiedavis, although this is an eslint rule update/code refactor, it's still quite a lot of files.

Mind taking the site for a spin with the devtools open to see if anything blows up including the admin section? I would guess no, but it'd be good to do a sanity check as we've only recently started adding e2e tests. I'll do the same.

Thanks!

**Comment by katiedavis** [comment]
> Not a request change, just a spot check.
> 
> @katiedavis, although this is an eslint rule update/code refactor, it's still quite a lot of files.
> 
> Mind taking the site for a spin with the devtools open to see if anything blows up including the admin section? I would guess no, but it'd be good to do a sanity check as we've only recently started adding e2e tests. I'll do the same.
> 
> Thanks!

Yeah for sure! I've just run through everything again and it's looking good on my end.

**Review by nickytonline — APPROVED** [review]
> Yeah for sure! I've just run through everything again and it's looking good on my end.

@katiedavis, I did a sanity check as well and things seems to be behaving properly. 🚢

**Pull request merged**